cmd/rest-server/main.go

```diff
@@ -111,21 +115,59 @@ func runRoot(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+
+	listener, err := findListener(inetd, server.Listen)
+	if err != nil {
+		return fmt.Errorf("unable to listen: %w", err)
+	}
+
 	if !enabledTLS {
-		log.Printf("Starting server on %s\n", server.Listen)
-		err = http.ListenAndServe(server.Listen, handler)
+		err = http.Serve(listener, handler)
 	} else {
-
-		log.Println("TLS enabled")
-		log.Printf("Private key: %s", privateKey)
-		log.Printf("Public key(certificate): %s", publicKey)
-		log.Printf("Starting server on %s\n", server.Listen)
-		err = http.ListenAndServeTLS(server.Listen, publicKey, privateKey, handler)
+		log.Printf("TLS enabled, private key %s, pubkey %v", privateKey, publicKey)
+		err = http.ServeTLS(listener, handler, publicKey, privateKey)
 	}
 
 	return err
 }
 
+func findListener(inetd bool, addr string) (listener net.Listener, err error) {
+	if inetd {
+		log.Print("inetd mode activated")
+		return net.FileListener(os.Stdin)
+	}
+
+	// try systemd socket activation
+	listeners, err := activation.Listeners()
+	if err != nil {
+		panic(err)
+	}
+
+	switch len(listeners) {
```

Comment:
Perhaps make the systemd socket activation explicit with a `--systemd` switch?

Reply:
Why do you think that's needed? We can discover (from the environment) if we're run via systemd socket activation...

Reply:
In that case you will end up ignoring the address configuration and using systemd instead, which could surprise the user. Adding a `--systemd` flag also makes clear that some magic is happening here.

Reply:
On the other hand, rest-server does log that it is in systemd mode, so I guess this is fine,